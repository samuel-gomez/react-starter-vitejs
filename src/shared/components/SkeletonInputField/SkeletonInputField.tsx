import type { ReactNode } from 'react';
import Skeleton from 'shared/components/Skeleton';

const SkeletonInputField = ({ label, id }: { label: ReactNode; id?: string }) => (
  <div className="row af-form__group">
    <div className="col-md-2">
      <label htmlFor={id} className="af-form__group-label">
        {label}
      </label>
    </div>
    <Skeleton />
  </div>
);

export default SkeletonInputField;
