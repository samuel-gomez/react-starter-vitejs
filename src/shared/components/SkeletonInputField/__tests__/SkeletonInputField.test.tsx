import { render } from '@testing-library/react';
import SkeletonInputField from '..';

describe('<SkeletonInputField/>', () => {
  it('Render <SkeletonInputField/>', () => {
    const { asFragment } = render(<SkeletonInputField label="hello" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
