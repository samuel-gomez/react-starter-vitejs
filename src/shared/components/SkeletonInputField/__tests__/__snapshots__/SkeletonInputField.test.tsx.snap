// Vitest Snapshot v1

exports[`<SkeletonInputField/> > Render <SkeletonInputField/> 1`] = `
<DocumentFragment>
  <div
    class="row af-form__group"
  >
    <div
      class="col-md-2"
    >
      <label
        class="af-form__group-label"
      >
        hello
      </label>
    </div>
    <div
      aria-busy="true"
      aria-label="loader"
      class="af-skeleton"
      role="status"
    />
  </div>
</DocumentFragment>
`;
